classes/payment-processor.php

```diff
@@ -124,16 +151,33 @@ public static function handle_response( $order, $payment_status = null, $redirec
 			throw new Exception( 'Mismatching payment ID' );
 		}
 
+		$last_payment_status = $order->get_meta( '_zaver_last_payment_status' );
+		$new_status          = $payment_status->getPaymentStatus();
+		$can_transition      = self::can_transition( $new_status, $last_payment_status );
+
+		if ( ! $can_transition ) {
+			ZCO()->logger()->error(
+				"Received unhandled payment status from Zaver: cannot transition from '$last_payment_status' to '$new_status'",
+				array(
+					'orderId'       => $order->get_id(),
+					'paymentId'     => $payment_status->getPaymentId(),
+					'paymentStatus' => $payment_status->getPaymentStatus(),
+				)
+			);
```

Comment:
Question: Should this be logged as an error? I'm thinking if as you say in `Payment_Processor::can_transition` we had an issue with `pending_merchant_capture` triggering multiple times, then this would log that as an error, when its not really an error.

Would it be better to log this as a debug instead to not put something as an error in the log when it might not be?